File: frontend/src/types/property.ts
export interface LandParcelEntity {
  id: string;
  base_ulpin: string;
  state_code: string;
  district_code: string;
  subdistrict_code: string;
  village_code: string;
  parcel_number: string;
  ground_elevation_amsl: number;
  area_sqm: number;
  land_use_type: string;
  status: "ACTIVE" | "PENDING_SURVEY" | "DISPUTED";
}

export interface BuildingEntity {
  id: string;
  land_parcel_id: string;
  name: string;
  building_code: string;
  total_height_m: number;
  total_floors: number;
  total_towers: number;
  structural_type: string;
}

export interface TowerEntity {
  id: string;
  building_id: string;
  tower_number: string;
  name: string;
  floor_count: number;
  base_elevation_amsl: number;
}

export interface FloorEntity {
  id: string;
  tower_id: string;
  floor_number: number;
  floor_label: string;
  z_min_amsl: number;
  z_max_amsl: number;
  floor_height_m: number;
  unit_count: number;
}

export interface StrataUnitEntity {
  id: string;
  floor_id: string;
  unit_number: string;
  ulpin_3d: string;
  carpet_area_sqm: number;
  builtup_area_sqm: number;
  volume_cum: number;
  base_elevation_amsl: number;
  ceiling_elevation_amsl: number;
  usage_type: "RESIDENTIAL" | "COMMERCIAL" | "MIXED";
  status: "REGISTERED" | "PENDING_SURVEY" | "DISPUTED";
}

File: backend/app/schemas/property.py
import uuid
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class LandParcelBase(BaseModel):
    base_ulpin: str
    state_code: str
    district_code: str
    subdistrict_code: str
    village_code: str
    parcel_number: str
    ground_elevation_amsl: float
    area_sqm: float
    land_use_type: str = "RESIDENTIAL"


class LandParcelRead(LandParcelBase):
    id: uuid.UUID
    status: str
    model_config = ConfigDict(from_attributes=True)


class BuildingRead(BaseModel):
    id: uuid.UUID
    land_parcel_id: uuid.UUID
    name: str
    building_code: str
    total_height_m: float
    total_floors: int
    total_towers: int
    structural_type: str
    model_config = ConfigDict(from_attributes=True)


class TowerRead(BaseModel):
    id: uuid.UUID
    building_id: uuid.UUID
    tower_number: str
    name: str
    floor_count: int
    base_elevation_amsl: float
    model_config = ConfigDict(from_attributes=True)


class FloorRead(BaseModel):
    id: uuid.UUID
    tower_id: uuid.UUID
    floor_number: int
    floor_label: str
    z_min_amsl: float
    z_max_amsl: float
    floor_height_m: float
    unit_count: int
    model_config = ConfigDict(from_attributes=True)


class StrataUnitRead(BaseModel):
    id: uuid.UUID
    floor_id: uuid.UUID
    unit_number: str
    ulpin_3d: str
    carpet_area_sqm: float
    builtup_area_sqm: float
    volume_cum: float
    base_elevation_amsl: float
    ceiling_elevation_amsl: float
    usage_type: str
    status: str
    model_config = ConfigDict(from_attributes=True)
